tests/test_bring.py

```diff
@@ -0,0 +1,614 @@
+        """Test does_user_exist for known user."""
+        mocked.get(
+            "https://api.getbring.com/rest/bringusers?email=EMAIL",
+            status=200,
+        )
+        assert await bring.does_user_exist() is True
+
+    @pytest.mark.parametrize(
+        ("exception", "expected"),
+        [
+            (asyncio.TimeoutError, BringRequestException),
+            (aiohttp.ClientError, BringEMailInvalidException),
+        ],
+    )
+    async def test_request_exception(self, mocked, bring, exception, expected):
+        """Test request exceptions."""
+
+        mocked.get(
+            "https://api.getbring.com/rest/bringusers?email=EMAIL",
+            exception=exception,
+        )
+
+        with pytest.raises(expected):
+            await bring.does_user_exist("EMAIL")
+
+
+class TestLogin:
+    """Tests for login method."""
+
+    async def test_mail_invalid(self, mocked, bring):
+        """Test login with invalid e-mail."""
+        mocked.post(
+            "https://api.getbring.com/rest/v2/bringauth",
+            status=400,
+        )
+        expected = "Login failed due to bad request, please check your email."
+        with pytest.raises(BringAuthException, match=expected):
+            await bring.login()
+
+    async def test_unauthorized(self, mocked, bring):
+        """Test login with unauthorized user."""
+        mocked.post(
+            "https://api.getbring.com/rest/v2/bringauth",
+            status=401,
+            payload={"message": ""},
+        )
+        expected = "Login failed due to authorization failure, please check your email and password."
+        with pytest.raises(BringAuthException, match=expected):
+            await bring.login()
+
+    @pytest.mark.parametrize("status", [200, 401])
+    async def test_parse_exception(self, mocked, bring, status):
+        """Test parse exceptions."""
+        mocked.post(
+            "https://api.getbring.com/rest/v2/bringauth",
+            status=status,
+            body="not json",
+            content_type="application/json",
+        )
+
+        with pytest.raises(BringParseException):
+            await bring.login()
+
+    @pytest.mark.parametrize(
+        "exception",
+        [
+            asyncio.TimeoutError,
+            aiohttp.ClientError,
+        ],
+    )
+    async def test_request_exceptions(self, mocked, bring, exception):
+        """Test exceptions."""
+        mocked.post("https://api.getbring.com/rest/v2/bringauth", exception=exception)
+        with pytest.raises(BringRequestException):
+            await bring.login()
+
+    async def test_login(self, mocked, bring, monkeypatch):
+        """Test login with valid user."""
+
+        mocked.post(
+            "https://api.getbring.com/rest/v2/bringauth",
+            status=200,
+            payload=BRING_LOGIN_RESPONSE,
+        )
+
+        async def mocked_get_user_account(*args, **kwargs):
+            """Mock __get_user_account."""
```

Comment:
intentional double __?

Reply:
nope, not intentional, get_user_account is a public method. Fixed it.